File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistItem.java
package pl.revanmj.smspasswordnotifier.data;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = WhitelistItem.TABLE_NAME)
public class WhitelistItem {
    @Ignore
    static final String TABLE_NAME = "Senders";

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "sender_id")
    private int senderId;

    private String name;
    private String regex;

    public int getSenderId() {
        return senderId;
    }

    public void setSenderId(int senderId) {
        this.senderId = senderId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRegex() {
        return regex;
    }

    public void setRegex(String regex) {
        this.regex = regex;
    }
}

File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistItem.kt
package pl.revanmj.smspasswordnotifier.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = WhitelistItem.TABLE_NAME)
class WhitelistItem(
        @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "sender_id") var senderId: Int = -1,
        var name: String,
        var regex: String
) {
    companion object {
        @Ignore
        internal const val TABLE_NAME = "Senders"
    }
}
